### PR TITLE
update urls for kinetics dataset

### PR DESCRIPTION
**Pull request by sahilg06**
Resolves issue #5564

### CODE REVIEW

File: torchvision/datasets/kinetics.py
Comment (pmeier) [inline torchvision/datasets/kinetics.py:180]: 1. Use [`urllib.parse.quote`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) instead of a manual solution
2. Maybe I'm missing something here, but since we overwrite `list_video_urls` we should be able to use a context manager here. Plus, instead of using `readlines` and manually stripping the newline delimiter away later, we can read the whole file and use `.split_lines()` to have it handled for us.

```suggestion
        with open(split_url_filepath) as file:
            list_video_urls = [urllib.parse.quote(line) for line in file.read().splitlines()]
```

Comment (sahilg06) [reply torchvision/datasets/kinetics.py]: Okay, got it.

**Comment by sahilg06** [comment]
> Looks good to me - LGTM
> 
> Question, should we patch the `.csv` vs `.txt` issue as well?

Yes, we can. I have the .csv files (which I have uploaded [here.](https://github.com/pytorch/vision/issues/5564#issuecomment-1068243392) ) . May you upload them somewhere and provide me with corresponding URLs as code needs the URLs. So that I can update the URLs as well.

**Comment by sahilg06** [comment]
Now the only thing left is to update the URLs for val.csv and train.csv of the k600 dataset. I have already uploaded the correct files [here](https://github.com/pytorch/vision/issues/5564#issuecomment-1068243392). 
@bjuncek @pmeier May you please generate URLs corresponding to those files and provide me with the same. So that I can update them in the code.

**Comment by sahilg06** [comment]
Why these tests are failing? I made a very small change in the last commit.

**Comment by pmeier** [comment]
Tests fail due to #5635. It is unrelated to your PR. You can ignore them. We'll have a final look before merge that your PR doesn't break anything.

**Comment by sahilg06** [comment]
Thank you @pmeier.
Please also have a look at [this](https://github.com/pytorch/vision/pull/5578#issuecomment-1072088874) comment.
So that everything is perfect.

**Comment by pmeier** [comment]
@bjuncek is handling the PR. I was under the impression everything is done with his approval?

**Comment by sahilg06** [comment]
Yes, he has approved all the previous changes. Also, he asked me [here](https://github.com/pytorch/vision/pull/5578#pullrequestreview-913432762) to patch the .csv vs .txt issue as well in this PR.
So it's only possible if someone provides me the new URLs corresponding to the correct files.

**Comment by pmeier** [comment]
I'm out of the loop here. Please wait for @bjuncek's review.

**Comment by bjuncek** [comment]
Hi @sahilg06 

I've just got Joao to merge the new csv's into kinetics S3, so the old CSV's should now work, i.e. all of the following should now have the same format:
```
https://s3.amazonaws.com/kinetics/600/annotations/train.csv
https://s3.amazonaws.com/kinetics/600/annotations/val.csv
https://s3.amazonaws.com/kinetics/600/annotations/test.csv
https://s3.amazonaws.com/kinetics/400/annotations/train.csv
https://s3.amazonaws.com/kinetics/400/annotations/val.csv
https://s3.amazonaws.com/kinetics/400/annotations/test.csv
```

Could you please verify this, and if that's ok, no further action w.r.t. URLs needs to be done

**Comment by pmeier** [comment]
Cross-posting https://github.com/pytorch/vision/issues/5564#issuecomment-1072239648

> Just to clear some confusion for someone else reading this thread: When @bjuncek talks about "Joao", he doesn't mean @jdsgomes, but rather Joao Carreira operating under @kinetics-cvdf. The links were updated in https://github.com/cvdfoundation/kinetics-dataset/commit/48a523ea10ad9d286e4f6f2895910fc4737e1ac5.

**Comment by sahilg06** [comment]
> Hi @sahilg06
> 
> I've just got Joao to merge the new csv's into kinetics S3, so the old CSV's should now work, i.e. all of the following should now have the same format:
> 
> ```
> https://s3.amazonaws.com/kinetics/600/annotations/train.csv
> https://s3.amazonaws.com/kinetics/600/annotations/val.csv
> https://s3.amazonaws.com/kinetics/600/annotations/test.csv
> https://s3.amazonaws.com/kinetics/400/annotations/train.csv
> https://s3.amazonaws.com/kinetics/400/annotations/val.csv
> https://s3.amazonaws.com/kinetics/400/annotations/test.csv
> ```
> 
> Could you please verify this, and if that's ok, no further action w.r.t. URLs needs to be done

Thanks @bjuncek. These URLs are working!
I have changed .txt to .csv for the same. Now everything seems good.
We can finally merge this PR.

**Review by bjuncek — APPROVED** [review]
LGTM!
Thanks for the work @sahilg06 and sorry for the delays

**Comment by sahilg06** [comment]
@NicolasHug may you please review and merge it asap? I need to use this class in one of my work.

**Comment by NicolasHug** [comment]
@sahilg06 when running your initial command

```py
 ds = torchvision.datasets.Kinetics(root='ktest',frames_per_clip=100, num_classes='600', split='val', download=True, num_download_workers=1000, num_workers=2)
```

I'm getting 

```
Downloading https://s3.amazonaws.com/kinetics/600/val/k600_val_path.txt to ktest/files/k600_val_path.txt
38912it [00:00, 445546.97it/s]                                                                                         
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-2-a444dcfd311f> in <module>
----> 1 ds = torchvision.datasets.Kinetics(root='ktest',frames_per_clip=100, num_classes='600', split='val', download=True, num_download_workers=1000, num_workers=2)

~/dev/vision/torchvision/datasets/kinetics.py in __init__(self, root, frames_per_clip, num_classes, split, frame_rate, step_between_clips, transform, extensions, download, num_download_workers, num_workers, _precomputed_metadata, _video_width, _video_height, _video_min_dimension, _audio_samples, _audio_channels, _legacy)

~/dev/vision/torchvision/datasets/kinetics.py in download_and_process_videos(self)

~/dev/vision/torchvision/datasets/kinetics.py in _download_videos(self)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/context.py in Pool(self, processes, initializer, initargs, maxtasksperchild)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/pool.py in __init__(self, processes, initializer, initargs, maxtasksperchild, context)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/pool.py in _repopulate_pool(self)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/pool.py in _repopulate_pool_static(ctx, Process, processes, pool, inqueue, outqueue, initializer, initargs, maxtasksperchild, wrap_exception)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/process.py in start(self)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/context.py in _Popen(process_obj)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/popen_fork.py in __init__(self, process_obj)

~/.miniconda3/envs/pt/lib/python3.9/multiprocessing/popen_fork.py in _launch(self, process_obj)

OSError: [Errno 24] Too many open files
```

**Comment by sahilg06** [comment]
@NicolasHug when I tried running it locally, it gave me the same error as yours. Previously I was trying on colab, where it was working properly. Let me check.

**Comment by sahilg06** [comment]
@NicolasHug This error is not due to changes I made, even if I tried running the command using the **Kinetics** class from **main** branch, it gave me the same error.
I think it's because of the large value of num_download_workers, I tried reducing it to 10, it worked perfectly.
So does this error depends upon the system, we are using?

**Comment by NicolasHug** [comment]
Thanks @sahilg06

**Pull request merged**
